src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -39,8 +43,31 @@
      * Андрей - 600
      * София - 900</p>
      */
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
-            throws Exception {
-        return null;
+    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
+        final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        LocalDate fromDate = LocalDate.parse(dateFrom, formatter);
+        LocalDate toDate = LocalDate.parse(dateTo, formatter);
+        if (fromDate.compareTo(toDate) > 0) {
+            throw new IllegalDateParametersException("Wrong parameters");
+        }
+        StringBuilder sb = new StringBuilder();
+        sb.append(String.format("Отчёт за период %s - %s\n", dateFrom, dateTo));
+        for (int i = 0; i < names.length; i++) {
+            sb.append(names[i]);
+            int salary = 0;
+            for (int j = 0; j < data.length; j++) {
+                String[] strings = data[j].split(" ");
+                boolean flag = names[i].equals(strings[1])
```

Comment:
instead of creating additional variable just use if construction

Reply:
done